### ISSUE TITLE
[Question] Should it take longer to open a populated database comparing to an empty one? 

### ISSUE THREAD

**Issue by drbolsen** [dfahlander/Dexie.js#1335]
I am just trying to figure out whether it is within specs that opening an existing database with some data takes significantly longer than an empty or lightly populated database?

I have an existing database with about 28 tables, most of the tables have about 5 to 10 rows except one that has about 5K records.  I did a simple measuring, nothing fancy: 

`   this.$db = new Dexie('existing')`
`   this.$db.version(1).stores({...skipped...})`
`   const t0 = performance.now() `
`   await this.$db.open() `
`   console.log('opening database [db]  took: ', (performance.now() - t0) + ' ms')`

that measured in average 435 - 500 ms

Doing the same measurement for an empty database
`   this.$db1 = new Dexie('whatever')`
`   this.$db1.version(1).stores({})`
`   const t1 = performance.now() `
`   await this.$db1.open() `
`   console.log('opening database [db1] took: ', (performance.now() - t1) + ' ms')`

resulted in average 20-25ms.

I've seen few comments around that the opening operation incurs  some time penalty, this is fine, but I wonder wether 20x difference is expected ?

**Comment by dfahlander** [comment]
Dexie itself doesn't do anything with existing data on open, so if there is this extra time, it is a plain browser issue. It is interesting to hear about this. Do you see a similar difference on Chrome, Safari and Firefox?

**Comment by drbolsen** [comment]
Thanks Davis, much appreciated the prompt response. I kind of felt that this should be the case 

I don't have Firefox installed at the moment, the application I am working on is a cross-platform app based on nwjs therefore I use Chrome/Chromium. Chrome Version 91.0.4472.101 (Official Build) (x86_64)

However,  I tried to run the app in Safari Version 14.1.1 (14611.2.7.1.6) on my Mac, the results are similar.

**Comment by drbolsen** [comment]
I took some liberty to play with my setup and the results are more puzzling to me. 

In a first round I commented the lines that open the existing database. Immediately the opening time for an empty database jumped from 20 ms to a whooping 400 ms. 

Furthermore I  added a call to IndexedDB API directly e.g. await window.indexedDB.open('existing') . Not sure wether this is the correct way of doing that but nevertheless it took just a miserable 0.2 ms!

I went further and disabled the lines that open the empty database, to my surprise the opening time in direct API call went from zero-ish value to another whooping jump of 70 ms.

I have no clue what is going on. The only thing that I can think of is that first ever access to IndexedDB API has time penalty regardless what database size is.

**Comment by dfahlander** [comment]
If I get your conclusion right, the size of the DB doesn't matter - only that the first ever call to indexedDB.open() takes about 400 ms no matter if it is empty or large. Is my understanding correct?

**Comment by drbolsen** [comment]
That's correct David, I can't find other explanation here. My application setup and tests are pretty straightforward, I don't use any plugins or hooks that I could blame.  I imagine that when IndexedDB is called first time ever it would go through some file operations, directory access, etc for the particular browser session/window. Once it is done the consequent operations are faster. 

Kind of disappointing, now it appears that I have  to initialise Dexie as soon as possible in my code.
